### PR TITLE
[IMP] stock: count all tracked products

### PR DESCRIPTION
**Pull request by kasp-odoo**
**[IMP] stock: count all tracked products**
> When request a count of quants, if some of them are for tracked product, also request a count for other quants for the same product in the same location.

Enterprise PR: https://github.com/odoo/enterprise/pull/49397
TaskId-3479636

### CODE REVIEW

File: addons/stock/wizard/stock_request_count.py [hunk 1]
```diff
@@ -18,9 +18,14 @@ class StockRequestCount(models.TransientModel):
 
     def action_request_count(self):
         for count_request in self:
+            if self.env.user.has_group('stock_barcode.group_barcode_count_entire_location'):
```
Comment (svs-odoo) [inline addons/stock/wizard/stock_request_count.py:21]: Should be done in enterprise since it depends of `stock_barcode`

**Comment by kasp-odoo** [comment]
@svs-odoo as per your suggestion above code has been moved to enterprise PR, https://github.com/odoo/enterprise/pull/49397

File: addons/stock/wizard/stock_request_count.py [hunk 2]
```diff
@@ -18,10 +19,30 @@ class StockRequestCount(models.TransientModel):
...
+            if quant.product_id.tracking == 'none':
+                continue
+            quant_domain = ['&',
+                            ('product_id', '=', quant.product_id.id),
+                            ('location_id', '=', quant.location_id.id)]
+            domain = expression.OR([domain, quant_domain]) if domain else quant_domain
+        if domain:
+            sibling_quants = self.env['stock.quant'].search(domain)
+            quants_to_count |= sibling_quants
+        return quants_to_count
```
Comment (Whenrow) [inline addons/stock/wizard/stock_request_count.py:45]: use a set to create a domain without duplicates

Comment (kasp-odoo) [reply addons/stock/wizard/stock_request_count.py]: Okay got it, changes are done.

Comment (svs-odoo) [reply addons/stock/wizard/stock_request_count.py]: @kasp-odoo That wasn't what is wanted... The domain itself is always a list, the set was for the location ids so we avoid duplicates in the domain (avoid to have multiple domain parts looking for the same product in the same location)
Anyway, I'll do it my side

**Pull request closed without merging**